### DISCUSSION TITLE
Count and filter seems gives strange count

### DISCUSSION THREAD

**Discussion by louisrli** [prisma/prisma#2934] [Q&A] [answered]
## Bug description
Unclear whether this is just me not understanding filter but I've read this documentation.

`StudySchedule` and `ScheduleDay` have a many-to-many relation.

Actual count: Some number in the thousands (like 4600)
Expected count: 35

Code:
```
    const numDays = await prisma.scheduleDay.count({
  1       where: {
  2         StudySchedule: {
  3           every: {
  4             id: schedule.id,
  5           },
  6         },
  7       },
  8     });
 12     console.log(`num days on schedule: ${numDays}`);
 13
```

```
prisma=> select count(*) from "_ScheduleDays" where "B" = 'ck9q2t6ddoxfe0726h0q5804c';
 count
-------
    35
(1 row)
```

Prisma is running this query:
```
prisma:query SELECT COUNT(*) FROM (SELECT "default$default"."ScheduleDay"."id" FROM "default$default"."ScheduleDay" WHERE ("default$default"."ScheduleDay"."id") NOT IN (SELECT "t0"."A" FROM "default$default"."_ScheduleDays" AS "t0" INNER JOIN "default$default"."StudySchedule" AS "j0" ON ("j0"."id") = ("t0"."B") WHERE (NOT "j0"."id" = $1)) OFFSET $2) AS "sub" LIMIT $3
```

I'm looking at [this documentation](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/filtering#retrieve-all-user-records-where-all-posts-are-published-and-a-least-one-related-post-mentions-prisma), by the way.

From the docs:
```
const result = await prisma.user.findMany({
    where: {
      post: {
          every: {
            published: true
          }
```

In English: Retrieve all User records where all posts are published

And so thus, I would assume that my query means:
"Count all schedule days where [all the] schedule IDs are equal to the provided ID"

Could you please clarify if I'm interpreting the documentation wrong or this is a bug? 

## How to reproduce
Not sure, I haven't done a small test case.

## Expected behavior

I expected it to yield the result corresponding to the English description I provided above.

## Prisma information
postgres, latest prisma

## Environment & setup

latest prisma, postgres 11

**Comment by pantharshit00** [comment, marked as answer]
Every key in the conditional works like Javascript every methods on Arrays. It makes sure that every subrecord will fulfills the condition and then it returns desired record.

So in your case, it will return only if there is one sub record of `ScheduleDay` which has the given id as I am assuming you can't connect in the database level. 


> select count(*) from "_ScheduleDays" where "B" = 'ck9q2t6ddoxfe0726h0q5804c';

The query that is fetching all the records connected which are connected with given id. It does now consider that we are fetching only if one record if the given id is present.

You will need `some` if you want to return record where one of the sub records have that id.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
